Library/ViewModels/ConfirmDetailsViewModel.swift

```diff
@@ -314,14 +314,22 @@ public class ConfirmDetailsViewModel: ConfirmDetailsViewModelType, ConfirmDetail
     let pledgeDetailsData = Signal.combineLatest(
       project,
       rewards,
+      selectedQuantities,
       pledgeTotal,
       refTag
     )
 
     let createCheckoutEvents = pledgeDetailsData
       .takeWhen(self.continueCTATappedProperty.signal)
-      .map { project, rewards, pledgeTotal, refTag in
-        let rewardsIDs = rewards.first?.isNoReward == true ? [] : rewards.map { $0.graphID }
+      .map { project, rewards, selectedQuantities, pledgeTotal, refTag in
+        let rewardsIDs: [String] = rewards.first?.isNoReward == true
+          ? []
+          : rewards.flatMap { reward -> [String] in
+            guard let count = selectedQuantities[reward.id] else {
+              return []
+            }
+            return [String](repeating: reward.graphID, count: count)
+          }
```

Comment:
might be a good idea to add a test for this

Reply:
Discussed offline - yeah, I'd be nice, but our test suite isn't designed to test that network calls are constructed correctly. So I'm going to just leave this for now.